File: project/frontend/src/model/product.ts
import {Manufacturer} from "./manufacturer.ts";
import {Category} from "./category.ts";
import {Stock} from "./stock.ts";

export interface Product {
    id: number,
    price: number,
    name: String,
    manufacturer: Manufacturer,
    manufacturingDate: Date,
    expirationDate: Date,
    category: Category,
    amount: number,
    stock: Stock
}

File: project/backend/work02/src/main/kotlin/br/ufc/work02/domain/model/mongo/Product.kt
package br.ufc.work02.domain.model.mongo

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.DBRef
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document("product")
data class Product(
    @Id
    var id: Long? = null,

    var name: String,

    var price: Double,

    @DBRef
    var manufacturer: Manufacturer,

    var manufacturingDate: Date,

    var expirationDate: Date,

    @DBRef
    var category: Category,

    var amount: Int,

    @DBRef
    var stock: Stock
)
